client/deviceCache.go

```diff
@@ -0,0 +1,196 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/plgd-dev/device/client/core"
+	"github.com/plgd-dev/go-coap/v2/pkg/cache"
+	"go.uber.org/atomic"
+)
+
+type DeviceCache struct {
+	defaultCacheExpiration time.Duration
+	devicesCache           *cache.Cache
+	errors                 func(error)
+
+	closed atomic.Bool
+	done   chan struct{}
+}
+
+func NewDeviceCache(defaultCacheExpiration, interval time.Duration, errors func(error)) *DeviceCache {
```

Comment:
Explain the functionality of deviceCacheExpiration e.g. the expiration is refreshed when the device is in use.

Reply:
fixed